src/core/trtserver.cc

```diff
@@ -626,6 +626,45 @@ TrtServerResponse::OutputData(
   return nullptr;  // Success
 }
 
+//
+// TrtServerModelIndex
+//
+// Implementation for TRTSERVER2_ModelIndex.
+//
+class TrtServerModelIndex {
+ public:
+  TrtServerModelIndex(ni::ModelRepositoryIndex model_repository_index);
+  TRTSERVER_Error* GetModelNames(
+      const char* const** models, uint64_t* models_count);
+
+ private:
+  std::vector<const char*> index_;
+};
+
+TRTSERVER_Error*
+TrtServerModelIndex::GetModelNames(
+    const char* const** models, uint64_t* models_count)
+{
+  if (index_.empty()) {
+    *models_count = 0;
+    *models = nullptr;
+  } else {
+    *models_count = index_.size();
+    *models = &(index_[0]);
+  }
+
+  return nullptr;
+}
+
+TrtServerModelIndex::TrtServerModelIndex(
```

Comment:
Move this constructor implementation above GetModelNames. Also, you still need to keep a copy of model_repository_index in the object, otherwise all your name pointers are going to become invalid once the called deletes that object. So just add " : model_repository_index_(model_repository_index)"